lib/pulse/lib/Pulse.Lib.Slice.fst

```diff
@@ -0,0 +1,219 @@
+module Pulse.Lib.Slice
+
+module AP = Pulse.Lib.ArrayPtr
+
+noeq
+type slice t = {
+    elt: AP.ptr t;
+    len: SZ.t;
+    fp: AP.footprint t;
+}
+
+let len s = s.len
+
+let pts_to #t s #p v =
+    AP.pts_to s.elt #p s.fp v **
+    pure (Seq.length v == SZ.v s.len)
+
+let pts_to_is_slprop2 x p v = ()
+
+```pulse
+ghost
+fn pts_to_len (#t: Type) (s: slice t) (#p: perm) (#v: Seq.seq t)
+requires    (pts_to s #p v)
+ensures    (pts_to s #p v ** pure (Seq.length v == SZ.v (len s)))
+
+{
+    unfold (pts_to s #p v);
+    fold (pts_to s #p v)
+}
+```
+
+let is_from_array #t a p s =
+    AP.is_from_array p s.fp a **
+    pure (SZ.v s.len == A.length a)
+
+```pulse
+fn from_array (#t: Type) (mutb: bool) (a: array t) (#p: perm) (#v: Ghost.erased (Seq.seq t)) (alen: SZ.t {
```

Comment:
What's the purpose of the `mutb` arguments here and in `split`?

Was your intention to encode the Rust `split`/`split_mut` distinction this way?  I don't think that works.  The main challenge with the Rust extraction is figuring out whether the type should be `&[t]` or `&mut[t]`; once you know the type it's easy to call the right `split(_mut)` version, but the flag in this function is not enough to infer the Rust type.